src/Controls/src/Core/Handlers/Items/iOS/SelectableItemsViewController.cs

```diff
@@ -57,6 +57,26 @@ internal void SelectItem(object selectedItem)
 				// Ensure the selected index is updated after the collection view's items generation is completed
 				CollectionView.PerformBatchUpdates(null, _ =>
 				{
+					// Exit if the ItemsSource reference no longer matches the one captured at invocation.
+					if (!ReferenceEquals(ItemsView.ItemsSource, originalSource))
```

Comment:
@bhavanesh2001  This issues could likely be resolved by using UICollectionView native calls instead of relying on 3+ managed method calls with object comparisons. I hope below fix will handle the edge cases where the item source is null or index position is removed or changed after the batch updates call. 

@PureWeen What do you think?

<img width="760" alt="image" src="https://github.com/user-attachments/assets/b2acfce7-f747-44b1-955c-be30bdab1a35" />